<!DOCTYPE html>
<html lang="en" itemscope itemtype="http://schema.org/WebPage">
  <head>
    

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0">

  <title>Command line version - System Manuals</title>
  <meta name="description" content="An Accurate and Ultra-deep Coverage Method for Large-scale SSR Genotyping with SNPs in the SSR and Flanking Region Compatible">
  <meta name="author" content="Maize Research Center"/><script type="application/ld+json">
{
    "@context": "http://schema.org",
    "@type": "WebSite",
    "name": "System Manuals",
    
    "url": "https:\/\/plantdna.github.io\/"
}
</script><script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "Organization",
  "name": "",
  "url": "https:\/\/plantdna.github.io\/"
  
  
  
  
}
</script>
<script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "BreadcrumbList",
  "itemListElement": [{
        "@type": "ListItem",
        "position": 1,
        "item": {
          "@id": "https:\/\/plantdna.github.io\/",
          "name": "home"
        }
    },{
        "@type": "ListItem",
        "position": 3,
        "item": {
          "@id": "https:\/\/plantdna.github.io\/post\/command-line\/",
          "name": "Command line version"
        }
    }]
}
</script><script type="application/ld+json">
{
  "@context": "http://schema.org",
  "@type": "Article",
  "author": {
    "name" : "huoyongxue"
  },
  "headline": "Command line version",
  "description" : "",
  "inLanguage" : "en",
  "wordCount":  393 ,
  "datePublished" : "2021-07-08T08:39:10",
  "dateModified" : "2021-07-08T08:39:10",
  "image" : "https:\/\/plantdna.github.io\/img\/avatar-icon.png",
  "keywords" : [ "AMGT-TS" ],
  "mainEntityOfPage" : "https:\/\/plantdna.github.io\/post\/command-line\/",
  "publisher" : {
    "@type": "Organization",
    "name" : "https:\/\/plantdna.github.io\/",
    "logo" : {
        "@type" : "ImageObject",
        "url" : "https:\/\/plantdna.github.io\/img\/avatar-icon.png",
        "height" :  60 ,
        "width" :  60
    }
  }
}
</script>

<meta property="og:title" content="Command line version" />
<meta property="og:description" content="An Accurate and Ultra-deep Coverage Method for Large-scale SSR Genotyping with SNPs in the SSR and Flanking Region Compatible">
<meta property="og:image" content="https://plantdna.github.io/img/avatar-icon.png" />
<meta property="og:url" content="https://plantdna.github.io/post/command-line/" />
<meta property="og:type" content="website" />
<meta property="og:site_name" content="System Manuals" />

  <meta name="twitter:title" content="Command line version" />
  <meta name="twitter:description" content="An Accurate and Ultra-deep Coverage Method for Large-scale SSR Genotyping with SNPs in the SSR and Flanking Region Compatible">
  <meta name="twitter:image" content="https://plantdna.github.io/img/avatar-icon.png" />
  <meta name="twitter:card" content="summary" />
  <link href='https://plantdna.github.io/img/favicon.ico' rel='icon' type='image/x-icon'/>
  <meta name="generator" content="Hugo 0.78.2" />
  <link rel="alternate" href="https://plantdna.github.io/index.xml" type="application/rss+xml" title="System Manuals"><link rel="stylesheet" href="https://plantdna.github.io/css/katex.min.css" />
  <link rel="stylesheet" href="https://plantdna.github.io/fontawesome/css/all.css" />
  <link rel="stylesheet" href="https://plantdna.github.io/css/bootstrap.min.css" /><link rel="stylesheet" href="https://plantdna.github.io/css/main.css" /><link rel="stylesheet" href="https://plantdna.github.io/css/fonts.css" />
  <link rel="stylesheet" href="https://plantdna.github.io/css/highlight.min.css" /><link rel="stylesheet" href="https://plantdna.github.io/css/codeblock.css" /><link rel="stylesheet" href="https://plantdna.github.io/css/photoswipe.min.css" />
  <link rel="stylesheet" href="https://plantdna.github.io/css/photoswipe.default-skin.min.css" />




  </head>
  <body>
    <nav class="navbar navbar-default navbar-fixed-top navbar-custom">
  <div class="container-fluid">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#main-navbar">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="https://plantdna.github.io/">System Manuals</a>
    </div>

    <div class="collapse navbar-collapse" id="main-navbar">
      <ul class="nav navbar-nav navbar-right">
        
          
            <li>
              <a title="Manual" href="/">Manual</a>
            </li>
          
        
          
            <li class="navlinks-container">
              <a class="navlinks-parent">Systems</a>
              <div class="navlinks-children">
                
                  <a href="/post/amgt-ts">AMGT-TS</a>
                
              </div>
            </li>
          
        
          
            <li>
              <a title="About" href="/page/about/">About</a>
            </li>
          
        
          
            <li>
              <a title="Tags" href="/tags">Tags</a>
            </li>
          
        

        
          
            <li>
              
                
              
                
                  <a href="/zh-cn" lang="zh-cn">zh-cn</a>
                
              
            </li>
          
        

        
      </ul>
    </div>

    
      <div class="avatar-container">
        <div class="avatar-img-border">
          <a title="System Manuals" href="https://plantdna.github.io/">
            <img class="avatar-img" src="https://plantdna.github.io/img/avatar-icon.png" alt="System Manuals" />
          </a>
        </div>
      </div>
    

  </div>
</nav>




    


<div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

<div class="pswp__bg"></div>

<div class="pswp__scroll-wrap">
    
    <div class="pswp__container">
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
      <div class="pswp__item"></div>
    </div>
    
    <div class="pswp__ui pswp__ui--hidden">
    <div class="pswp__top-bar">
      
      <div class="pswp__counter"></div>
      <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>
      <button class="pswp__button pswp__button--share" title="Share"></button>
      <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>
      <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>
      
      
      <div class="pswp__preloader">
        <div class="pswp__preloader__icn">
          <div class="pswp__preloader__cut">
            <div class="pswp__preloader__donut"></div>
          </div>
        </div>
      </div>
    </div>
    <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
      <div class="pswp__share-tooltip"></div>
    </div>
    <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)">
    </button>
    <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)">
    </button>
    <div class="pswp__caption">
      <div class="pswp__caption__center"></div>
    </div>
    </div>
    </div>
</div>


  
  
  






  
    <div id="header-big-imgs" data-num-img=1 
      
         
          data-img-src-1="/img/path.jpg" 
         
         data-img-desc-1="Path"
      ></div>
  

  <header class="header-section has-img">
    
      <div class="intro-header big-img">
        
        <div class="container">
          <div class="row">
            <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1">
              <div class="post-heading">
                <h1>Command line version</h1>
                  
                    
                      <h2 class="post-subheading">An Accurate and Ultra-deep Coverage Method for Large-scale SSR Genotyping with SNPs in the SSR and Flanking Region Compatible</h2>
                    
                  
                  
                    <span class="post-meta">
  
  
  <i class="fas fa-calendar"></i>&nbsp;
  
  
    &nbsp;|&nbsp;<i class="fas fa-clock"></i>&nbsp;2&nbsp;minutes
  
  
    &nbsp;|&nbsp;<i class="fas fa-book"></i>&nbsp;393&nbsp;words
  
  
  
</span>


                  
              </div>
            </div>
          </div>
        </div>
        <span class="img-desc" style="display: inline;"></span>
      </div>
    
    <div class="intro-header no-img">
      <div class="container">
        <div class="row">
          <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1">
            <div class="post-heading">
              
                <h1>Command line version</h1>
              
              
              
                
                  <h2 class="post-subheading">An Accurate and Ultra-deep Coverage Method for Large-scale SSR Genotyping with SNPs in the SSR and Flanking Region Compatible</h2>
                
              
              
                <span class="post-meta">
  
  
  <i class="fas fa-calendar"></i>&nbsp;
  
  
    &nbsp;|&nbsp;<i class="fas fa-clock"></i>&nbsp;2&nbsp;minutes
  
  
    &nbsp;|&nbsp;<i class="fas fa-book"></i>&nbsp;393&nbsp;words
  
  
  
</span>


              
            </div>
          </div>
        </div>
      </div>
    </div>
  </header>


    
<div class="container" role="main">
  <div class="row">
    <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1">
      <article role="main" class="blog-post">
        <h1 id="amgt-ts-user-manual-command-line-version">AMGT-TS User manual (command line version)</h1>
<ol>
<li>Prerequisite
Running AMGT-TS requires a GNU-like environment. It should be possible to run AMGT-TS on a Linux or Mac OS, Ubuntu Server 18.04 is recommended.</li>
</ol>
<p>AMGT-TS requires the following external tools:</p>
<div class="highlight"><pre class="chroma"><code class="language-fallback" data-lang="fallback">bamtools　(2.5.0)
blast tool suite (2.6.0+)
bwa　(0.7.17-r1188)
fastx_toolkit (0.0.13)
picard　(2.15.0-SNAPSHOT)
samtools (1.3.1)
seqtk (1.2)
</code></pre></div><p>So, please install the external tools before the following steps.</p>
<ol start="2">
<li>Get the source code</li>
</ol>
<p>git clone <a href="https://github.com/plantdna/amgt-ts.git">https://github.com/plantdna/amgt-ts.git</a></p>
<ol start="3">
<li>Add the samples data and the loci information file
For test purpose, just skip this step, for we already provided two sample data files under the folder &ldquo;working/00_fastq&rdquo;:
artificial-reads-5000-2loci.fq
sample15.fq</li>
</ol>
<p>and loci information files under folder &ldquo;ref&rdquo;:
sites.amplicon.fa
sites.motif.info.stats</p>
<ol start="4">
<li>Setup the configuration file
A example configuration file is under the &ldquo;subtools&rdquo; folder: profiles-maizedna-1.sh</li>
</ol>
<p>partial lines as:</p>
<div class="highlight"><pre class="chroma"><code class="language-fallback" data-lang="fallback">export TOOLS_DIR=/mnt/diskc/gitlab/s2s/tools

export FORMAT=fq

export PICARD=$TOOLS_DIR/picard.jar
</code></pre></div><p>Please change the value same as the environment.</p>
<ol start="5">
<li>Run</li>
</ol>
<p>Although we can run the tool as the description in the README.md file, we recommend using &ldquo;launch.sh&rdquo; instead. For it will bring some convenient. Let&rsquo;s have a look of this script:</p>
<div class="highlight"><pre class="chroma"><code class="language-bash" data-lang="bash"><span class="cp">#!/bin/bash
</span><span class="cp"></span>
<span class="c1"># Keep all running information to log file.</span>
<span class="nv">TIME_STR</span><span class="o">=</span><span class="sb">`</span>date +<span class="s2">&#34;%Y%m%d%H%M%S&#34;</span><span class="sb">`</span>
<span class="nv">LOG_FILE</span><span class="o">=</span>run_<span class="nv">$TIME_STR</span>.log
touch <span class="nv">$LOG_FILE</span>
<span class="nv">SCRIPT_DIR</span><span class="o">=</span>/mnt/diskc/gitlab/amgt-ts/
<span class="nv">ENV_FILE</span><span class="o">=</span><span class="nv">$SCRIPT_DIR</span>/subtools/profiles-maizedna-1.sh
<span class="c1">#METHOD=broad</span>
<span class="nv">METHOD</span><span class="o">=</span>precise
<span class="nv">PROJECT_DIR</span><span class="o">=</span>/mnt/diskc/gitlab/amgt-ts
<span class="nv">$SCRIPT_DIR</span>/amgt-ts.sh --script<span class="o">=</span><span class="nv">$SCRIPT_DIR</span> --environment<span class="o">=</span><span class="nv">$ENV_FILE</span> --method<span class="o">=</span><span class="nv">$METHOD</span> --project<span class="o">=</span><span class="nv">$PROJECT_DIR</span> 2&gt;<span class="p">&amp;</span><span class="m">1</span> <span class="p">|</span> tee -a <span class="nv">$LOG_FILE</span>
</code></pre></div><p>The following 4 lines need to be changed:</p>
<div class="highlight"><pre class="chroma"><code class="language-fallback" data-lang="fallback">SCRIPT_DIR=/mnt/diskc/gitlab/amgt-ts/
ENV_FILE=$SCRIPT_DIR/subtools/profiles-maizedna-1.sh
METHOD=precise
PROJECT_DIR=/mnt/diskc/gitlab/amgt-ts
</code></pre></div><p>After modify this script to consistent with the environment, let&rsquo;s start with just input:
$ ./launch.sh</p>
<p>We can get the genotype result under folder &ldquo;working/04_reads&rdquo;, such as:</p>
<div class="highlight"><pre class="chroma"><code class="language-fallback" data-lang="fallback">$ head -n 3 working/04_reads/sample15.site.stat
#Site Motif Repeat_len Reads_num Total_reads Proportion
s258878 GCT 0 10 2370 0.0042194092827
s258878 GCT 12 1 2370 0.00042194092827


$ head -n 3 working/04_reads/artificial-reads-5000-2loci.site.stat
#Site Motif Repeat_len Reads_num Total_reads Proportion
s258878 GCT 9 1000 1000 1.0
s282049 TGG 15 1000 2000 0.5
</code></pre></div><p>and the reads of each genotype:</p>
<div class="highlight"><pre class="chroma"><code class="language-fallback" data-lang="fallback">$ head -n 6 working/04_reads/sample15/s258878/SSR9.fas
&gt;CRC8E:03524:00917
GATCTGTTTGCCAGCTGGGGCAAGATTTCTTCGGTGCCTGCACCCTTTCTCTTGCGGTTGTTTATGCTATCGCTGCTGCTGATTGTGGGGTTCCTGCGTTCGCCACTGTGACTGTCACTTTGCTGGTGCTGTTCCTGGTTGCATCTGCTTTTCAGTATGTGGGGCTTGAGCTTGTTC
&gt;CRC8E:03306:03509
TAACTGTGCCTTGATCTGTTTGCCAGCTGGGGCAAGATTTCTTCGGTGCCTGCACCCTTTCTCTTGCGGTTGTTTATGCTATCGCTGCTGCTGATTGTGGGGTTCCTGCGTTCGCCACTGTGACTGTCACTTTGCTGGTGCTGTTCCTGGTTGCATCTGCTTTTCAGTATGTGGGGCTTGAGCTTGTTC
&gt;CRC8E:04520:05520
GATCTGTTTGCCAGCTGGGGCAAGATTTCTTCGGTGCCTGCACCCTTTCTCTTGCGGTTGTTTATGCTATCGCTGCTGCTGATTGTGGGGTTCCTGCGTTCGCCACTGTGACTGTCACTTTGCTGGTGCTGTTCCTGGTTGCATCTGCTTTTCAGTATGTGGGGCTTGAGCTTGTTC
</code></pre></div><p>If you encounter some issue, you can check the log file:</p>
<div class="highlight"><pre class="chroma"><code class="language-fallback" data-lang="fallback">$ head -n 5 run_20210707191020.log
===== Script starts running on Wed Jul 7 19:10:20 CST 2021 =====
[bwa_index] Pack FASTA... 0.00 sec
[bwa_index] Construct BWT for the packed sequence...
[bwa_index] 0.00 seconds elapse.
[bwa_index] Update BWT... 0.00 sec
</code></pre></div><p>Note that the log file is named by time, so change the file name same with the log file in your tool folder.</p>
<p>Any questions, feel free to contact via <a href="mailto:holibut@gmail.com">email</a>, thank you very much to have interest in this tool.</p>

        
          <div class="blog-tags">
            
              <a href="https://plantdna.github.io//tags/amgt-ts/">AMGT-TS</a>&nbsp;
            
          </div>
        

        

        
          
            
          

          
                  <h4 class="see-also">See also</h4>
                  <ul>
                
                
                    <li><a href="/post/web-based/">Web-based version</a></li>
                
              </ul>

          
        
      </article>

      
        <ul class="pager blog-pager">
          
            <li class="previous">
              <a href="https://plantdna.github.io/post/web-based/" data-toggle="tooltip" data-placement="top" title="Web-based version">&larr; Previous Post</a>
            </li>
          
          
            <li class="next">
              <a href="https://plantdna.github.io/post/input-file-format/" data-toggle="tooltip" data-placement="top" title="AMGT-TS input file format">Next Post &rarr;</a>
            </li>
          
        </ul>
      


      
        
        
      

    </div>
  </div>
</div>

      
<footer>
  <div class="container">
    <div class="row">
      <div class="col-lg-8 col-lg-offset-2 col-md-10 col-md-offset-1">
        <ul class="list-inline text-center footer-links">
          
              <li>
                <a href="mailto:youremail@domain.com" title="Email me">
                  <span class="fa-stack fa-lg">
                    <i class="fas fa-circle fa-stack-2x"></i>
                    <i class="fas fa-envelope fa-stack-1x fa-inverse"></i>
                  </span>
                </a>
              </li>
              <li>
                <a href="https://github.com/plantdna/amgt-ts" title="GitHub">
                  <span class="fa-stack fa-lg">
                    <i class="fas fa-circle fa-stack-2x"></i>
                    <i class="fab fa-github fa-stack-1x fa-inverse"></i>
                  </span>
                </a>
              </li>
          
        </ul>
        <p class="credits copyright text-muted">
          
            
              Maize Research Center
            
          

          &nbsp;&bull;&nbsp;&copy;
          
            2021
          

          
            &nbsp;&bull;&nbsp;
            <a href="https://plantdna.github.io/">System Manuals</a>
          
        </p>
        
        <p class="credits theme-by text-muted">
          <a href="https://gohugo.io">Hugo v0.78.2</a> powered &nbsp;&bull;&nbsp; Theme <a href="https://github.com/halogenica/beautifulhugo">Beautiful Hugo</a> adapted from <a href="https://deanattali.com/beautiful-jekyll/">Beautiful Jekyll</a>
          
        </p>
      </div>
    </div>
  </div>
</footer><script src="https://plantdna.github.io/js/katex.min.js"></script>
<script src="https://plantdna.github.io/js/auto-render.min.js"></script>
<script src="https://plantdna.github.io/js/jquery.min.js"></script>
<script src="https://plantdna.github.io/js/bootstrap.min.js"></script>

<script src="https://plantdna.github.io/js/main.js"></script>
<script src="https://plantdna.github.io/js/highlight.min.js"></script>
<script> hljs.initHighlightingOnLoad(); </script>
<script> $(document).ready(function() {$("pre.chroma").css("padding","0");}); </script><script> renderMathInElement(document.body); </script><script src="https://plantdna.github.io/js/photoswipe.min.js"></script>
<script src="https://plantdna.github.io/js/photoswipe-ui-default.min.js"></script><script src="https://plantdna.github.io/js/load-photoswipe.js"></script>









    
  </body>
</html>

